src/TextUI/TestRunner.php

```diff
@@ -1145,6 +1147,13 @@ private function processSuiteFilters(TestSuite $suite, array $arguments): void
             );
         }
 
+        if (!empty($arguments['testsXml'])) {
```

Comment:
Around here is the place to load+parse your configuration and get a list of tests. Perhaps you can even adapt/reuse the current `NameFilterIterator`, as it's basically what it is. :)